### PR TITLE
mon/MDSMonitor: set birth time on FSMap during encode

### PR DESCRIPTION
**Pull request by batrick**
See also: https://github.com/rook/rook/issues/12789

So we can begin to answer questions like: when did we last see an MDS?

Fixes: https://tracker.ceph.com/issues/62849

TODO:
- ~docs~
- ~tests~
- ~release note~
- ~add command "last-seen <mds>"~

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
So I think this works now. Just need to add testing.

File: src/pybind/mgr/dashboard/openapi.yaml [hunk 1]
```diff
@@ -4306,6 +4306,9 @@ paths:
                                 epoch:
                                   description: ''
                                   type: integer
+                                btime:
+                                  description: ''
```
[the remark this answers is not in the document]
Comment (batrick) [reply src/pybind/mgr/dashboard/openapi.yaml]: I don't know. Perhaps the dashboard team just never got around to documenting these fields. (I don't see a reason to start in this PR!) :)

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65597](https://tracker.ceph.com/issues/65597).

File: src/pybind/mgr/dashboard/openapi.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vshankar) [reply src/pybind/mgr/dashboard/openapi.yaml:4985]: Fair enough!

**Comment by batrick** [comment]
cc @BlaineEXE

**Comment by batrick** [comment]
Sorry this got put on the back burner for a while. It's ready now.

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/66433](https://tracker.ceph.com/issues/66433).

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test dashboard

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/66462](https://tracker.ceph.com/issues/66462).

**Comment by batrick** [comment]
jenkins test dashboard

**Comment by batrick** [comment]
jenkins test dashboard cephadm

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test dashboard

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test dashboard cephadm

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test dashboard

**Comment by batrick** [comment]
jenkins test dashboard cephadm

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test dashboard

**Comment by batrick** [comment]
jenkins test dashboard cephadm

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test dashboard cephadm

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/66609](https://tracker.ceph.com/issues/66609).

**Comment by batrick** [comment]
Ready for merge. @ceph/cephfs @vshankar requires approval please.

**Review by vshankar — APPROVED** [review]
Nice work!

**Pull request merged**